meegpowreg/pipelines.py

```diff
@@ -17,126 +17,300 @@
+
+    Prepare filter bank models as used in [1]_. These models take as input
+    sensor-space covariance matrices computed from M/EEG signals in different
+    frequency bands. Then transformations are applied to improve the
+    applicability of linear regression techniques by reducing the impact of
+    field spread.
+
+    In terms of implementation, this involves 1) projection
+    (e.g. spatial filters) and 2) vectorization (e.g. taking the log on the
+    diagonal).
+
+    .. note::
+        The resulting model expects as inputs data frames in which different
+        covarances (e.g. for different frequencies) are stored inside columns
+        indexed by ``names``.
+
+        Other columns will be passed through by the underlying column
+        transformers.
+
+        The pipeline also supports fitting categorical interaction effects
+        after projection and vectorization steps are performed.
+
+    .. note::
+        All essential methods from [1]_ are implemented here. In practice,
+        we recommend comparing `riemann', `spoc' and `diag' as a baseline.
+
+    Parameters
+    ----------
+    names : list of str
+        The column names of the data frame corresponding to different
+        covariances.
+    method : str
+        The method used for extracting features from covariances. Defaults
+        to ``'riemann'``.
+    projection_params : dict
+        The parameters for the projection step.
+    vectorization_params : dict
+        The parameters for the vectorization step.
+    categorical_interaction : str
+        The column in the input data frame containing a binary descriptor
+        used to fit 2-way interaction effects.
+    scaling : scikit-learn Transformer object | None
+        Method for re-rescaling the features. Defaults to None. If None,
+        StandardScaler is used.
+    estimator : scikit-learn Estimator object.
+        The estimator object. Defaults to None. If None, LogisticRegression
+        is performed with default values.
+
+    References
+    ----------
+    [1] D. Sabbagh, P. Ablin, G. Varoquaux, A. Gramfort, and D.A. Engemann.
+        Predictive regression modeling with MEG/EEG: from source power
+        to signals and cognitive states.
+        *NeuroImage*, page 116893,2020. ISSN 1053-8119.
+        https://doi.org/10.1016/j.neuroimage.2020.116893
+
+    """
+    filter_bank_transformer = make_filter_bank_transformer(
+        names=names, method=method, projection_params=projection_params,
+        vectorization_params=vectorization_params,
+        categorical_interaction=categorical_interaction
     )
-    expander = make_column_transformer(
-        *[(ExpandFeatures(expand=expand), [col, -1])
-          for col in range(len(fb_cols))],
-        ('drop', -1)
+
+    scaling_ = scaling
+    if scaling_ is None:
+        scaling_ = StandardScaler()
+
+    estimator_ = estimator
+    if estimator_ is None:
+        estimator_ = LogisticRegression()
```

Comment:
```suggestion
        estimator_ = LogisticRegression(solver='liblinear')
```
always faster from my experience